db/seeds.rb

```diff
@@ -68,161 +68,196 @@ def fact_check_attributes(fact_check_link, user, project, team)
-        'clients': [
-        {
-            'id': random_string,
-            'status': 'active',
-            'externalId': phone,
-            'active': true,
-            'lastSeen': Time.now.to_s,
-            'platform': 'whatsapp',
-            'integrationId': random_string,
-            'displayName': phone,
-            'raw': {
-            'profile': {
-                'name': tipline_user_name
-            },
-            'from': phone
-            }
-        }
-        ],
-        'pendingClients': []
-    },
-    'identifier': random_string,
-    'app_name': random_string
-    }
-
-    fields = {
-    smooch_user_id: uid,
-    smooch_user_app_id: random_string,
-    smooch_user_data: smooch_user_data.to_json
-    }
-
-    Dynamic.create!(annotation_type: 'smooch_user', annotated: team, annotator: BotUser.smooch_user, set_fields: fields.to_json)
-
-    # Tipline request
-    smooch_data = {
-    'role': 'appUser',
-    'source': {
-        'type': 'whatsapp',
-        'id': random_string,
-        'integrationId': random_string,
-        'originalMessageId': random_string,
-        'originalMessageTimestamp': Time.now.to_i
-    },
-    'authorId': uid,
-    'name': tipline_user_name,
-    '_id': random_string,
-    'type': 'text',
-    'received': Time.now.to_f,
-    'text': tipline_text,
-    'language': 'en',
-    'mediaUrl': nil,
-    'mediaSize': 0,
-    'archived': 3,
-    'app_id': random_string
-    }
-
-    fields = {
-    smooch_request_type: 'default_requests',
-    smooch_data: smooch_data.to_json
-    }
-
-    a = Dynamic.new(annotation_type: 'smooch', annotated: project_media, annotator: BotUser.smooch_user)
-    a.set_fields = fields.to_json
-    a.save!
+puts "If you want to create a new user: press 1 then enter"
+puts "If you want to add more data to an existing user: press 2 then enter"
+print ">> "
+answer = STDIN.gets.chomp
+
+ActiveRecord::Base.transaction do
+  if answer == "1"
+    puts 'Making Team / Workspace...'
+    team = create_team(name: Faker::Company.name)
+
+    puts 'Making User...'
+    user = create_user(name: data[:user_name], login: data[:user_name], password: data[:user_password], password_confirmation: data[:user_password], email: Faker::Internet.safe_email(name: data[:user_name]), is_admin: true)
+
+    puts 'Making Project...'
+    project = create_project(title: team.name, team_id: team.id, user: user, description: '')
+
+    puts 'Making Team User...'
+    create_team_user(team: team, user: user, role: 'admin')
+  elsif answer == "2"
+    puts "Type user email then press enter"
+    print ">> "
+    email = STDIN.gets.chomp
+
+    puts "Fetching User, Project, Team User and Team..."
+    user = User.find_by(email: email)
+
+    if TeamUser.find_by(user_id: user.id).nil?
+      team = create_team(name: Faker::Company.name)
+      project = create_project(title: team.name, team_id: team.id, user: user, description: '')
+      create_team_user(team: team, user: user, role: 'admin')
```

Comment:
@hartsick I'm wondering, a user might not have a team, but might have a project. However, do we want to give the option to add to a project? Just for generating fake data isn't creating a project enough?

Reply:
@vasconsaurus honestly, I'm not sure I'm familiar enough with the data model to understand the difference between project / team / team_user and what we'd need. would you mind describing them to me? my inclination is that whatever is minimum amount of relations necessary is enough for the seeds file

Reply:
Team is your workspace and project is the folder, so in the image bellow we are in the project Walsh-Flatley inside the team Walsh-Flatley (I don't remember why I named them the same thing 😅 ). As far as I understand team_user just connects those two.

<img width="1392" alt="Screenshot 2023-05-04 at 20 00 00" src="https://user-images.githubusercontent.com/87862340/236348023-6a20a6ad-0357-4f60-ad1a-baea151b11fe.png">

Reply:
ah gotcha! thanks for explaining!